x-pack/plugins/asset_manager/server/routes/assets.ts

```diff
@@ -23,11 +26,29 @@ const assetType = schema.oneOf([
 const getAssetsQueryOptions = schema.object({
   from: schema.maybe(schema.string()),
   to: schema.maybe(schema.string()),
-  type: schema.maybe(schema.oneOf([schema.arrayOf(assetType), assetType])),
+  type: schema.maybe(schema.oneOf([schema.arrayOf(assetTypeRT), assetTypeRT])),
   ean: schema.maybe(schema.oneOf([schema.arrayOf(schema.string()), schema.string()])),
   size: schema.maybe(schema.number()),
 });
 
+const getAssetsDiffQueryOptions = schema.object({
+  aFrom: schema.string(),
+  aTo: schema.string(),
+  bFrom: schema.string(),
+  bTo: schema.string(),
+  type: schema.maybe(schema.oneOf([schema.arrayOf(assetType), assetType])),
+});
+
+const getRelatedAssetsQueryOptions = schema.object({
+  from: schema.string(), // ISO timestamp or ES datemath
+  to: schema.maybe(schema.string()), // ISO timestamp or ES datemath
+  ean: schema.string(),
+  relation: relationRT,
+  type: schema.maybe(schema.oneOf([assetTypeRT, schema.arrayOf(assetTypeRT)])),
+  maxDistance: schema.maybe(schema.number()),
+  size: schema.maybe(schema.number()),
```

Comment:
Nice tips, I'll add them to the iots issue here https://github.com/elastic/kibana/issues/154147.
Regarding the dates I was leaning towards only accepting epoch or iso since it's easy to validate and conversion for clients is trivial enough. I'm wondering if es datemath are commonly accepted in kibana apis or if we expect the translation to happen before (maybe through the datepicker for ui-based calls)

Reply:
I think we dug into this once in a pairing session and for example the Kibana time service translates to ISO internally if you give it a datemath string, so the requests sent are in ISO so I think we should follow that, meaning, the API should reject if it's not ISO format date string.